### ISSUE TITLE
image-similarity app succeed in non-pip env but failed in pip env 

### ISSUE THREAD

**Issue by glorysdj** [intel-analytics/analytics-zoo#2220]
non-pip succeed 
#6 image-similarity time used:2475 seconds
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Python-AppTests-part1/12/consoleFull

pip failed
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Pip-Apps-Integration-Part1/11/consoleFull

```bash
Start evaluation:
Traceback (most recent call last):
  File "/opt/work/jenkins/workspace/ZOO-PR-Pip-Apps-Integration-Part1/dist/apps/image-similarity/tmp.py", line 177, in <module>
    predictionDF.sample(False, 0.01).show()
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/pyspark/sql/dataframe.py", line 378, in show
    print(self._jdf.showString(n, 20, vertical))
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/py4j/java_gateway.py", line 1257, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/pyspark/sql/utils.py", line 63, in deco
    return f(*a, **kw)
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/py4j/protocol.py", line 336, in get_return_value
    format(target_id, ".", name))
py4j.protocol.Py4JError: An error occurred while calling o182.showString
Traceback (most recent call last):
  File "/opt/work/conda/envs/py36/lib/python3.6/socketserver.py", line 320, in _handle_request_noblock
    self.process_request(request, client_address)
  File "/opt/work/conda/envs/py36/lib/python3.6/socketserver.py", line 351, in process_request
    self.finish_request(request, client_address)
  File "/opt/work/conda/envs/py36/lib/python3.6/socketserver.py", line 364, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/opt/work/conda/envs/py36/lib/python3.6/socketserver.py", line 724, in __init__
    self.handle()
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/pyspark/accumulators.py", line 269, in handle
    poll(accum_updates)
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/pyspark/accumulators.py", line 241, in poll
    if func():
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/pyspark/accumulators.py", line 245, in accum_updates
    num_updates = read_int(self.rfile)
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/pyspark/serializers.py", line 717, in read_int
    raise EOFError
EOFError
----------------------------------------
ERROR:py4j.java_gateway:An error occurred while trying to connect to the Java server (127.0.0.1:39786)
Traceback (most recent call last):
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/py4j/java_gateway.py", line 929, in _get_connection
    connection = self.deque.pop()
IndexError: pop from an empty deque

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/py4j/java_gateway.py", line 1067, in start
    self.socket.connect((self.address, self.port))
ConnectionRefusedError: [Errno 111] Connection refused
ERROR:py4j.java_gateway:An error occurred while trying to connect to the Java server (127.0.0.1:39786)
Traceback (most recent call last):
  File "/opt/work/conda/envs/py36/lib/python3.6/site-packages/py4j/java_gateway.py", line 929, in _get_connection
    connection = self.deque.pop()
```

**Comment by glorysdj** [comment]
fixed

**Comment by hkvision** [comment]
Is expanding driver memory the fix? Is #2222 still necessary?

**Comment by glorysdj** [comment]
2222 is great. we can fix this with both.
